### ISSUE TITLE
Improved Music Rooms

### ISSUE THREAD

**Issue by Xelitik** [tetrio/issues#432]
### Checklist (issue will be disregarded & closed if incomplete!)

- [x] I've **checked both [open and recently closed issues](https://github.com/tetrio/issues/issues?q=is%3Aissue)** to ensure this feature is not already requested.
- [x] I am using the **newest version** of TETR.IO.
- [x] I am certain this is a **feature request**, not a bug.
- [x] I've **thought this feature through** properly.

-----

### Is your feature request related to a problem? Please describe.
When talking about the Music Room, I am referring to the menus such as
About > Music Room
Solo > 40 Lines > [Song]
Multiplayer > Custom Game > Music: [Song]
Etc.

When you select a song in the Music Room, the list of songs disappear, and you have to reopen the Music Room. This would not be much of a problem if the list did not start from the top after reopening. This makes it a little tedious if you want to listen, look through, or choose from the currently 50 songs that you want to use. Which itself is a lot to look through in the first place.

### Describe the solution you'd like
The room will be improved by not closing out when selecting a song. It would be simple for users to choose a song then hit the back button afterwards.

It would be nice if the songs were color coded to differ from the type of song:
Random: Default Color
Interface: Color 1 (a vivid purple/pink?)
Calm: Color 2 (a soft blue?)
Special: Color 3 (yellow?)
Battle: Color 4 (red?)

Issue #321 would also give the About Music Room a good use, though you should still be able to add favorites in other Music Rooms, with a [Random: Favorite] option added.

### Additional context
Mock-up of the color coded songs and concept favorite system. (If added hopefully the favorite button will have a better placement.)

![imr_issue_ac1](https://user-images.githubusercontent.com/62867685/87473304-10938d00-c5d6-11ea-82e5-cac6afcb5353.png)

**Comment by Xelitik** [comment]
This issue is somewhat dated, so here is the current gist:

- Having songs colored based on it's type would make selection much easier.
- Sorting by song type, player rating (frequency), artist could make this even easier, but is not necessary.
- Song previews could start on more recognizable parts, playing when hovering over a song (similar to osu!).
- Favoriting can still be useful alongside the current frequency option by allowing players to sort by favorites without conflicting with their frequency ratings.

**Comment by othello777** [comment]
Indeed! I've only played for a couple of days and I had the idea where if you hover over a song it previews it. that would help a lot.
